lang/java/tools/src/main/java/org/apache/avro/tool/DataFileReadTool.java

```diff
@@ -84,8 +94,27 @@ public int run(InputStream stdin, PrintStream out, PrintStream err,
     return 0;
   }
 
+  private static long getHeadCount(OptionSet optionSet, OptionSpec<String> headOption, List<String> nargs) {
+    long headCount = Long.MAX_VALUE;
+    if(optionSet.has(headOption)) {
+      headCount = DEFAULT_HEAD_COUNT;
+      List<String> headValues = optionSet.valuesOf(headOption);
+      if(headValues.size() > 0) {
+        // if the value parses to int, assume it's meant to go with --head
+        // otherwise assume it was an optionSet.nonOptionArgument and add back to the list
+        // TODO: support input filenames whose whole path+name is int parsable?
```

Comment:
isn't there a flag that can separate optional args from the nonOptionArgument? something like `--head -- 1`